core/server/common/src/main/java/alluxio/RpcUtils.java

```diff
@@ -144,6 +145,43 @@ public static <T> T callAndReturn(Logger logger, RpcCallableThrowsIOException<T>
     }
   }
 
+  /**
+   * Calls the given method and handled exception. Exceptions are
+   * accounted for in metrics and then rethrown at the end.
+   *
+   * @param logger           the logger to use for this call
+   * @param future           the future to call
+   * @param methodName       the name of the method, used for metrics
+   * @param description      the format string of the description, used for logging
+   * @param responseObserver gRPC response observer
+   * @param args             the arguments for the description
+   * @param <T>              the return type of the method
+   */
+  public static <T> void invoke(Logger logger, CompletableFuture<T> future, String methodName,
+      String description, StreamObserver<T> responseObserver, Object... args) {
+    // avoid string format for better performance if debug is off
+    String debugDesc =
+        logger.isDebugEnabled() ? String.format(description, processObjects(logger, args)) : null;
+    try (MetricsSystem.MultiTimerContext ctx = new MetricsSystem.MultiTimerContext(
+        MetricsSystem.timer(MetricKey.MASTER_TOTAL_RPCS.getName()),
+        MetricsSystem.timer(getQualifiedMetricName(methodName)))) {
+      MetricsSystem.counter(getQualifiedInProgressMetricName(methodName)).inc();
+      logger.debug("Enter: {}: {}", methodName, debugDesc);
+      future.whenComplete((r, t) -> {
```

Comment:
@rongrong I change this function to take a future so I think we can reuse this one in the future as well. Also notice that here we can use whenCompleteAsync to pass in worker rpc thread as an alternative. I feel like whenComplete should be fine in most cases.